### PR TITLE
Add a script to generate documentation

### PR DESCRIPTION
**Pull request by Palleas**
I figured it would be cool to have a nice generated documentation. 
TL;DR: this is what it looks like for now:

screen shot 2016-07-27 at 10 09 15 pm

Since we need the source to generate the documentation, I create this script. It's a little weird 
but it does the job right know and I'm more than opened to making changes. The idea is to: 
- Generate the documentation using jazzy
- Remove the current `gh-pages` since it's only for generated documentation, we don't really care
  about the history
- Checkout to a new, orphan `gh-pages` branch and cleanup every files besides `docs/`
- Move the content of `docs` to the root directory
- Stage and commit everything

I was curious as to how other open source projects manage to generate documentation but I 
couldn't find anything so... 

What do you think? 

(I'll probably squash everything once we validate this (or another) approach. I'm glad I did because
my first attempt deleted the content of my .git folder 😨.

### CODE REVIEW

**Comment by mdiep** [comment]
Very nice! I like it!

**Comment by Palleas** [comment]
I'll see if there is a better way to manage that but I don't really see how... I could do everything on the `gh-pages` branch and start by downloading an archive of the latest master..? I'd say it's a good enough start, I'll probably add more doc soon and push on the gh-pages for the 0.4 version once this is merged. I would have liked some feedback on my super dirty script but my call on twitter was left unanswered 😁

**Comment by Palleas** [comment]
I have found a much cleaner way, thanks to robb/Cartography#128. The `gh-pages` has a submodule to Tentacle's master and a script to rebuild the documentation properly. 
It looks pretty cool: http://mdiep.github.io/Tentacle/index.html 

Please let me know what you think 😄

**Comment by mdiep** [comment]
That looks great!

**Comment by Palleas** [comment]
Cool, I'll close this and add something in the readme then. 🙂
